metadata/src/main/java/edu/unc/lib/dl/metrics/GaugeFactory.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.metrics;
+
+import io.dropwizard.metrics5.Gauge;
+import io.dropwizard.metrics5.MetricName;
+import io.dropwizard.metrics5.MetricRegistry;
+
+/**
+ * A factory for creating gauges for reporting various metrics
+ *
+ * @author harring
+ *
+ */
+public class GaugeFactory {
+
+    private static final MetricRegistry registry = RegistryService.getInstance().getRegistry();
+
+    private GaugeFactory() {
+
+    }
+
+    /**
+     * Creates and registers a gauge under the given name
+     *
+     * @param metricName the name under which to register the gauge
+     * @param duration the length of time elapsed in ms
+     */
+    public static void createDurationGauge(MetricName name, Long duration) {
```

Comment:
I think these signatures are good for now. I do wonder if they should be named something like "incrementDurationGauge" instead though since they don't actually return an instance of the gauge.

Reply:
A bit more generally, I'm wondering if we should be using histograms for this purpose? The description in their documentation is rather sparse, but my impression is it is for taking measurements of occurrences:
https://metrics.dropwizard.io/3.1.0/getting-started/#histograms
I found section 3.4 in here a little clearer:
http://www.baeldung.com/dropwizard-metrics

Reply:
Re: your first comment, while it's true that no Gauge is returned by the method, nevertheless one is created and registered by the registry. For this reason I do like "create" in the signature. There really isn't any incrementing per se that happens with a gauge, but rather just the recording of a value for a particular deposit.

As for your more general concern, I guess it depends on how we want to use the data. Are we more interested in recording individual durations for deposits, or do we want them all to appear in aggregate in a histogram to make it easier to see trends?